src/ngraph/builder/reshape.cpp

```diff
@@ -177,10 +178,28 @@ shared_ptr<Node> builder::expand_dims(const Output<Node>& value, size_t axis)
         ->add_provenance_group_members_above({value});
 }
 
+NGRAPH_API
 shared_ptr<Node> builder::opset1::reshape(const Output<Node>& value, const Shape& shape)
 {
-    const auto out_pattern = op::Constant::create(
-        element::i64, Shape{shape.size()}, vector<int64_t>(shape.begin(), shape.end()));
+    shared_ptr<op::Constant> out_pattern;
+    if (value.get_shape() == shape)
```

Comment:
I suggest using `value.get_output_partial_shape(0).same_scheme(shape)` instead to avoid exceptions when `value`'s shape is not static.